mlflow/tracking/fluent.py

```diff
@@ -582,8 +583,20 @@ def log_param(key: str, value: Any) -> Any:
     :param value: Parameter value (string, but will be string-ified if not).
                   All backend stores support values up to length 500, but some
                   may support larger values.
-
-    :return: the parameter value that is logged.
+    :param synchronous: [Experimental] Defaults to True.
+                        Indicates if the parameter would be logged in synchronous
+                            fashion or not.
+                        When it is True this call would be blocking call and offers immediate
+                            consistency of the parameter upon returning.
+                        When this value is set to False, parameter would be logged in async
+                            fashion. Backing provider gurantees that metrics are accepted
+                            into system but would persist them with some time delay -
+                            'near-real time'/'eventual' consistency.
```

Comment:
I think we should simplify this:

```suggestion
        :param synchronous: *Experimental* If True, blocks until the metric is logged successfully. If False,
                                             logs the metric asynchronously and returns a future representing the logging operation.
```

Reply:
Can we apply this change to the other methods in `fluent.py` too?